gexpect.go

```diff
@@ -71,7 +71,7 @@ func (buf *buffer) ReadRune() (r rune, size int, err error) {
 		if err != nil {
 			return 0, 0, err
 		}
-		if utf8.FullRune(chunk) {
+		if utf8.FullRune(chunk[:l]) {
```

Comment:
I would use `n` instead of `l` here - it does not matter in practice if `l` or `n` is used here, but it is easier to reason about this code when we case see that `n` being tied to the `chunk` variable and `l` to the `buf.b` member, instead of mixing them like that.